src/v/cluster/tm_stm.cc

```diff
@@ -24,14 +24,19 @@
 
 namespace cluster {
 
-template<typename T>
-static model::record_batch serialize_cmd(T t, model::record_batch_type type) {
-    storage::record_batch_builder b(type, model::offset(0));
-    iobuf key_buf;
-    reflection::adl<uint8_t>{}.to(key_buf, T::record_key);
-    iobuf v_buf;
-    reflection::adl<T>{}.to(v_buf, std::move(t));
-    b.add_raw_kv(std::move(key_buf), std::move(v_buf));
+static model::record_batch serialize_tx(tm_transaction tx) {
+    iobuf key;
+    reflection::serialize(key, tm_update_batch_type());
```

Comment:
will these batches be in a kafka log? if so, should we special case the handling based on type in the compaction layer since we can't control the keys from clients?

Reply:
in theory we should already support this because otherwise there might be a collision between configuration batches and data batches but I've created a ticket to check that different batch types have independent keyspaces - https://app.clubhouse.io/vectorized/story/2968 but we don't support compaction in the transactional workloads so it isn't the top priority issue